Sw diag

JHopeCollins opened the pull request:
These changes will update the nonlinear shallow water equations example to use the new implicit scheme for the block solves.

colinjcotter reviewed (commented):
Should I wait till the utils stuff is merged before commenting? Then we can use utils for this.

JHopeCollins commented:
> Should I wait till the utils stuff is merged before commenting? Then we can use utils for this.

Probably sensible. Once the utils branch is merged, I'll merge that into this branch and then we can review

colinjcotter commented:
Looks like you have yet to update the preconditioner options to MG?

colinjcotter commented:
ah sorry, missed that!
________________________________
From: JHopeCollins ***@***.***>
Sent: 12 May 2022 11:01
To: colinjcotter/asQ ***@***.***>
Cc: Cotter, Colin J ***@***.***>; Comment ***@***.***>
Subject: Re: [colinjcotter/asQ] Sw diag (PR #28)


Looks like you have yet to update the preconditioner options to MG?

In sw_diag.py?
I started a new example swim.py for the new implicit solve so that I kept the sw_diag script to compare back to. We can remove the sw_diag script now if its not needed.

—
Reply to this email directly, view it on GitHub<https://github.com/colinjcotter/asQ/pull/28#issuecomment-1124801775>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ABOSV4VCOSFX5PVIPJEJWDDVJTJIFANCNFSM5U67HWMA>.
You are receiving this because you commented.Message ID: ***@***.***>

JHopeCollins commented:
Once the utils branch is merged and I've tidied up the scripts with it, I'll do a bit of reorganisation so we have an examples/shallow_water directory with:
williamson2_linear.py
williamson2_nonlinear.py
williamson5_nonlinear.py

The other cases (for WCP1 or anything else we want) could go on a different pull request so that we can get this one merged in

JHopeCollins commented:
Changes made:
- The only changes in asQ are to add the transfer managers onto the blocks (although this still needs the debug for the Firedrake warning)
- I removed the linear_sw and sw_diag examples and replaced them with the williamson2 and williamson5 scripts that use the new implicit solver for the blocks
- williamson5_verify tests and prints out the L2 error for the velocity and depth compared to the serial solution using the same solve parameters as the block solves.
- The scripts have been simplified a bit with utils

colinjcotter reviewed (commented):
This all looks good. My main question is whether we can automate some verification through pytest. Possibly not, because it is too slow - unless we can pass in arguments to do the verification over a shorter time interval? What do you think?

JHopeCollins commented:
> This all looks good. My main question is whether we can automate some verification through pytest. Possibly not, because it is too slow - unless we can pass in arguments to do the verification over a shorter time interval? What do you think?

I think this is possible, although if we do that I think that separating out unit, integration, and verification tests would be useful. 
Possibly the github actions would only test the unit and integration tests to keep the time down.

JHopeCollins commented:
Verification test added for the williamson5 case.
Serial and parallel solves are in the utils.shallow_water module

JHopeCollins commented:
I think this is ready to go.

Since the last review I refactored the verification tests. There are now functions for serial and parallel solves of williamson5 in the utils.shallow_water submodule. These functions get called both in the verification test, and the verification script.

colinjcotter commented:
Happy with this!
